tests/unit/src/test/scala/tests/codeactions/FlatMapToForComprehensionSuite.scala

```diff
@@ -0,0 +1,167 @@
+       |         }
+       |        }
+       |
+       |}
+       |""".stripMargin,
+    selectedActionIndex = 1
+  )
+
+  check(
+    "flatten-for-flatMap",
+    """|object A {
+       |    def double(x : Int, y: Int = 1) = y * x
+       |    def check(x: Int) = true
+       |    val list = List(1, 2, 3)
+       |
+       |    val res3 = list
+       |        .flatMap{
+       |              a => {
+       |                        val m = 6
+       |                        Some(a + 1).map(b => b + 3 + 4)
+       |                    }
+       |
+       |        }.filter (check)
+       |        .filterNot(_ =>  true)
+       |        .map(_ => 7)
+       |        .map(c => c - 1)
+       |        .map( double(_, 4).toFloat.toInt)
+       |        .filter(d => d > 1)
+       |        .map(5 + double(_, 7).toFloat.toInt / 8 + 6)
+       |        .map( _.toInt.compare(3))
+       |        .fl<<>>atMap(  m => Some(m * 3))
+       |
+       |}
+       |""".stripMargin,
+    s"""|${RewriteBracesParensCodeAction.toBraces("flatMap")}
+        |${FlatMapToForComprehensionCodeAction.flatMapToForComprehension}
+        |""".stripMargin,
+    """|object A {
+       |    def double(x : Int, y: Int = 1) = y * x
+       |    def check(x: Int) = true
+       |    val list = List(1, 2, 3)
+       |
+       |    val res3 = {
+       |         for {
+       |           a <- list
+       |           generatedByMetals5 <- {
+       |             val m = 6
+       |             Some(a + 1).map(b => b + 3 + 4)
+       |           }
+       |           if check(generatedByMetals5)
+       |           generatedByMetals4 = generatedByMetals5
+       |           if !true
+       |           generatedByMetals3 = generatedByMetals4
+       |           c = 7
+       |           generatedByMetals2 = c - 1
+       |           d = double(generatedByMetals2, 4).toFloat.toInt
+       |           if d > 1
+       |           generatedByMetals1 = d
+       |           generatedByMetals0 = 5 + double(generatedByMetals1, 7).toFloat.toInt / 8 + 6
+       |           m = generatedByMetals0.toInt.compare(3)
+       |           generatedByMetals <- Some(m * 3)
+       |         }  yield {
+       |           generatedByMetals
+       |         }
+       |        }
+       |
+       |}
+       |""".stripMargin,
+    selectedActionIndex = 1
+  )
+
+//   writing checkNoAction does not work, since `RewriteBraces` action always shows up in this case.
+//   but writing it the way I have writtern also does not work, because the result has type errors.
+//
+//  check("double-replacement-placeHolder",
+//    """|object A{
+//       |def multiply(a: Int, b: Int) = a * b
+//       |val res = List(1, 2, 3).m<<>>ap(multiply(_, _))
+//      |}""".stripMargin,
+//    RewriteBracesParensCodeAction.toBraces("map"),
+//    """|object A{
+//       |def multiply(a: Int, b: Int) = a * b
+//       |val res = List(1, 2, 3).map{multiply(_, _)}
+//       |}""".stripMargin,
```

Comment:
```suggestion
//       |}""".stripMargin,
   expectError = true
```

Reply:
Thus test case fails for non-compilability of the result. Lets not add it. @dos65